int32 motor
int32 angle
int32 temperature